contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -1283,146 +1287,298 @@ sp_describe_undeclared_parameters_internal(PG_FUNCTION_ARGS)
+		", CASE T2.name COLLATE sys.database_default "
+		"WHEN \'image\' THEN -1 "
+		"WHEN \'ntext\' THEN -1 "
+		"WHEN \'text\' THEN -1 "
+		"ELSE CASE "
+				"WHEN a.atttypmod != -1 THEN "
+					"sys.tsql_type_max_length_helper(coalesce(T2.name, tsql_base_type_name), a.attlen, a.atttypmod) "
+				"ELSE "
+					"sys.tsql_type_max_length_helper(coalesce(T2.name, tsql_base_type_name), a.attlen, t.typtypmod) "
+			"END  "
 		"END  "					/* AS "suggested_max_length" */
-		", C.precision "		/* AS "suggested_precision" */
-		", C.scale "			/* AS "suggested_scale" */
-		", CASE WHEN T.user_type_id = T.system_type_id THEN CAST( NULL AS INT ) ELSE T.user_type_id END "	/* AS
+		", CASE "
+				"WHEN a.atttypmod != -1 THEN "
+					"sys.tsql_type_precision_helper(coalesce(T2.name, tsql_base_type_name), a.atttypmod) "
+				"ELSE "
+					"sys.tsql_type_precision_helper(coalesce(T2.name, tsql_base_type_name), t.typtypmod) "
+			"END "		/* AS "suggested_precision" */
+		", CASE "
+				"WHEN a.atttypmod != -1 THEN "
+					"sys.tsql_type_scale_helper(coalesce(T2.name, tsql_base_type_name), a.atttypmod, false) "
+				"ELSE "
+					"sys.tsql_type_scale_helper(coalesce(T2.name, tsql_base_type_name), t.typtypmod, false) "
+			"END "			/* AS "suggested_scale" */
+		", CASE WHEN T2.user_type_id = T2.system_type_id THEN CAST( NULL AS INT ) ELSE T2.user_type_id END "	/* AS
 																											 * "suggested_user_type_id" */
-		", CASE WHEN T.user_type_id = T.system_type_id THEN CAST( NULL AS sysname) ELSE DB_NAME() END " /* AS
+		", CASE WHEN T2.user_type_id = T2.system_type_id THEN CAST( NULL AS sysname) ELSE DB_NAME() END " /* AS
 																										 * "suggested_user_type_database" */
-		", CASE WHEN T.user_type_id = T.system_type_id THEN CAST( NULL AS sysname) ELSE SCHEMA_NAME( T.schema_id ) END "	/* AS
+		", CASE WHEN T2.user_type_id = T2.system_type_id THEN CAST( NULL AS sysname) ELSE SCHEMA_NAME( T2.schema_id ) END "	/* AS
 																															 * "suggested_user_type_schema" */
-		", CASE WHEN T.user_type_id = T.system_type_id THEN CAST( NULL AS sysname) ELSE T.name END "	/* AS
+		", CASE WHEN T2.user_type_id = T2.system_type_id THEN CAST( NULL AS sysname) ELSE T2.name END "	/* AS
 																										 * "suggested_user_type_name" */
 		", CAST( NULL AS NVARCHAR(4000) ) " /* AS
 											 * "suggested_assembly_qualified_type_name" */
-		", CASE "
-		"WHEN C.xml_collection_id = 0 THEN CAST( NULL AS INT ) "
-		"ELSE C.xml_collection_id "
-		"END "					/* AS "suggested_xml_collection_id" */
+		", CAST( NULL AS INT )  "				/* AS "suggested_xml_collection_id" */
 		", CAST( NULL AS sysname ) "	/* AS
 										 * "suggested_xml_collection_database" */
 		", CAST( NULL AS sysname ) "	/* AS
 										 * "suggested_xml_collection_schema" */
 		", CAST( NULL AS sysname ) "	/* AS "suggested_xml_collection_name" */
-		", C.is_xml_document "	/* AS "suggested_is_xml_document" */
+		", CAST(0 AS sys.bit) "	/* AS "suggested_is_xml_document" */
```

Comment:
why?

Reply:
is_xml_document column is hardcoded as 0 in sys.columns. If we have any way to obtain this value in future, then we have to update this definition as same as sys.columns_internal